validate/lab01_test.go

```diff
@@ -20,86 +20,80 @@ var _ = Describe("Lab 1 Containers", func() {
 	Context("Step 2", func() {
 		It("should have a Dockerfile", func() {
 			failMessage = "Dockerfile Doesn't Exist or is in the wrong location\n"
-			Expect(fileExists("../Dockerfile")).To(Succeed(), failMessage)
+			Expect("../Dockerfile").To(BeAnExistingFile(), failMessage)
 		})
 	})
 
 	Context("Step 3", func() {
 		It("should have a skaffold.yaml file", func() {
 			failMessage = "skaffold.yaml Doesn't Exist or is in the wrong location\n"
-			Expect(fileExists("../skaffold.yaml")).To(Succeed(), failMessage)
+			Expect("../skaffold.yaml").To(BeAnExistingFile(), failMessage)
 		})
 
-		It("should be a valid skaffold configuration", func() {
-			var skaffoldActual interface{}
-			var skaffoldExpected interface{}
-
-			skaffoldExpectedFile, err := ioutil.ReadFile("./solution-data/lab01step03-skaffold.yaml")
-			err = yaml.Unmarshal(skaffoldExpectedFile, &skaffoldExpected)
-
-			skaffoldActualFile, err := ioutil.ReadFile("../skaffold.yaml")
-			err = yaml.Unmarshal(skaffoldActualFile, &skaffoldActual)
-
-			err = treeCompare(skaffoldActual, skaffoldExpected)
-			Expect(err).To(BeNil())
-
-			var skaffold interface{}
-			skaffoldFile, err := ioutil.ReadFile("../skaffold.yaml")
-			if err != nil {
-				Skip("skaffold.yaml not found")
-			}
-
-			err = yaml.Unmarshal(skaffoldFile, &skaffold)
-			Expect(err).ToNot(HaveOccurred())
-			//failures := InterceptGomegaFailures(func() {
+		It("should have valid versions", func() {
+			skaffoldExpected := YamlToInterface("../skaffold.yaml")
+			skaffoldActual := YamlToInterface("./solution-data/lab01step03/skaffold-version.yaml")
+			Expect(skaffoldActual).To(ValidateYamlObject(skaffoldExpected))
```

Comment:
Maybe we could pass in the filenames and let ValidateYamlObject open and parse the files? That way it could also handle files that fail to parse but it might be easier for the messaging to handle "failed to parse" and "failed to validate structure" as separate tests.

Reply:
As far as I could tell, I don't think there was any way of passing in strings into ValidateYamlObject since the Gomega matcher expects an interface as an argument